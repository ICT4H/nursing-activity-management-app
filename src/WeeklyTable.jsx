import React, {Component} from 'react';
import moment from "moment/moment";
import PropTypes from 'prop-types';

class WeeklyTable extends Component {
  getCurrentWeekDates() {
    let currentWeekDates = [];
    let startingDate = this.props.currentWeek.startingDate;

    for (let i = 0; i < 7; i++) {
      currentWeekDates.push(<th key={'day' + i}>{moment(startingDate).day(i).toDate().toDateString()}</th>);
    }
    return currentWeekDates;
  }

  render() {
    let headerRow = [];
    headerRow.push(<th>{this.props.title ? this.props.title : ""}</th>);
    headerRow = headerRow.concat(this.getCurrentWeekDates());
    let resultantTableOfData = [];


    if (this.props.data && true) {
      resultantTableOfData = this.getRowForEachDataObject();
    }

    return (<div>
      <table>
        <tbody>
        <tr>{headerRow}</tr>
        {resultantTableOfData}
        </tbody>
      </table>
    </div>)
  }

  getRowForEachDataObject() {
    let startingDate = this.props.currentWeek.startingDate;
    let allRows = [];

    this.props.data.forEach((object) => {
      let row = [];
      row.push(<td>{object.getTitleOfRow()}</td>);
      for (let i = 0; i < 7; i++) {
        let date = moment(startingDate).day(i).toDate();
        row.push(<td>{object.getDetailsFor(date)}</td>);
      }
      allRows.push(<tr>{row}</tr>)
    });

    return allRows
  }
}


WeeklyTable.propTypes = {
  currentWeek: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  data: PropTypes.arrayOf(PropTypes.object),
};
export default WeeklyTable
