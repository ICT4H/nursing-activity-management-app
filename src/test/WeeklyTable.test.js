import React from 'react';
import WeeklyTable from '../WeeklyTable';
import renderer from 'react-test-renderer';
import moment from "moment";

test('Provided title and currentWeek having starting date should ' +
    'give table with row having title and dates of given week as table headers', () => {
  const currentWeek = {
    startingDate: moment().day(0)
  };
  let component=renderer.create(<WeeklyTable currentWeek={currentWeek} title={"Column1"}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Provided data as a list of objects having two methods as getTitleOfRow and getDetailsFor(givenDate)' +
    'should give table consisting row for each object', () => {
  const currentWeek = {
    startingDate: moment(new Date('June 5, 2018 9:30:00')).day(0).toDate()
  };
  let sampleObject={
    getTitleOfRow(){
      return "titleOfRow";
    },
    getDetailsFor(date){
      return date.toString();
    }
  };

  let data=[sampleObject];

  let component=renderer.create(<WeeklyTable currentWeek={currentWeek} title={"Column1"} data={data}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});