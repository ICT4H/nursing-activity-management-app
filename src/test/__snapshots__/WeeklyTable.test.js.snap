// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provided data as a list of objects having two methods as getTitleOfRow and getDetailsFor(givenDate)should give table consisting row for each object 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th>
          Column1
        </th>
        <th>
          Sun Jun 03 2018
        </th>
        <th>
          Mon Jun 04 2018
        </th>
        <th>
          Tue Jun 05 2018
        </th>
        <th>
          Wed Jun 06 2018
        </th>
        <th>
          Thu Jun 07 2018
        </th>
        <th>
          Fri Jun 08 2018
        </th>
        <th>
          Sat Jun 09 2018
        </th>
      </tr>
      <tr>
        <td>
          titleOfRow
        </td>
        <td>
          Sun Jun 03 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Mon Jun 04 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Tue Jun 05 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Wed Jun 06 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Thu Jun 07 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Fri Jun 08 2018 09:30:00 GMT+0530 (IST)
        </td>
        <td>
          Sat Jun 09 2018 09:30:00 GMT+0530 (IST)
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Provided title and currentWeek having starting date should give table with row having title and dates of given week as table headers 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th>
          Column1
        </th>
        <th>
          Sun Jun 03 2018
        </th>
        <th>
          Mon Jun 04 2018
        </th>
        <th>
          Tue Jun 05 2018
        </th>
        <th>
          Wed Jun 06 2018
        </th>
        <th>
          Thu Jun 07 2018
        </th>
        <th>
          Fri Jun 08 2018
        </th>
        <th>
          Sat Jun 09 2018
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
